swiftz_core/swiftz_core/Error.swift

```diff
@@ -7,8 +7,10 @@
 //
 
 /// Immediately terminates the program with an error message.
+///
+/// This currently only works in Debug mode, since the underlying assertion is optimized away in Release mode.
```

Comment:
Does Swift let us use __builtin_unreachable()?  That won't get optimized away in release.